File: app/src/main/java/com/softaai/designpatterns/java/creational/AbstractFactory/shape/ShapeFactory.java
package com.softaai.designpatterns.java.creational.AbstractFactory.shape;

import com.softaai.designpatterns.java.creational.AbstractFactory.AbstractFactory;
import com.softaai.designpatterns.java.creational.AbstractFactory.color.Color;

/**
 * Created by Amol Pawar on 27-03-2019.
 * softAai Apps
 */
public class ShapeFactory extends AbstractFactory {

    @Override
    protected Color getColor(String color) {
        return null;
    }

    public Shape getShape(String shapeType){
        if(shapeType == null){
            return null;
        }
        if(shapeType.equalsIgnoreCase("RECTANGLE")){
            return new Rectangle();
        }
        else if(shapeType.equalsIgnoreCase("SQUARE")){
            return new Square();
        }
        else if(shapeType.equalsIgnoreCase("CIRCLE")){
            return new Circle();
        }
        return null;
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/creational/AbstractFactory/shape/ShapeFactory.kt
package com.softaai.designpatterns.kotlin.creational.AbstractFactory.shape

import com.softaai.designpatterns.kotlin.creational.AbstractFactory.AbstractFactory
import com.softaai.designpatterns.kotlin.creational.AbstractFactory.color.Color


/**
 * Created by Amol Pawar on 01-04-2019.
 * softAai Apps
 */
class ShapeFactory : AbstractFactory() {
    override fun getColor(color: String): Color? {
        return null
    }

    override fun getShape(shape: String): Shape? {
        if(shape.equals("RECTANGLE", ignoreCase = true)){
            return Rectangle()
        }
        else if(shape.equals("SQUARE", ignoreCase = true)){
            return Square()
        }
        else if(shape.equals("CIRCLE", ignoreCase = true)){
            return Circle()
        }

        return null
    }
}